test/e2e/config_test.go

```diff
@@ -71,16 +71,33 @@ func TestClusterMonitoringOperatorConfiguration(t *testing.T) {
 	t.Log("asserting that CMO goes degraded after an invalid configuration is pushed")
 	f.AssertOperatorCondition(configv1.OperatorDegraded, configv1.ConditionTrue)(t)
 	f.AssertOperatorCondition(configv1.OperatorAvailable, configv1.ConditionFalse)(t)
+	// Even when an invalid configuration is caught by both unmarshallers, the operator is still set
+	// to Upgradeable=false, this keeps the logic simple and encourage users to fix their configs.
+	f.AssertOperatorCondition(configv1.OperatorUpgradeable, configv1.ConditionFalse)(t)
 	f.AssertOperatorConditionReason(configv1.OperatorDegraded, "InvalidConfiguration")
 	f.AssertOperatorConditionReason(configv1.OperatorAvailable, "InvalidConfiguration")
 	// Check that the previous setup hasn't been reverted
 	f.AssertStatefulsetExists("prometheus-user-workload", f.UserWorkloadMonitoringNs)(t)
 
+	t.Log("invalid configuration with field of the wrong case")
```

Comment:
no need have those as we don't want to propagate the same unmarshalling errors as warnings, see above.